frontend/src/hooks/useSearchToursAPI.ts

```diff
@@ -0,0 +1,12 @@
+import type { SearchResponse } from '@/api/api';
+import { searchToursAPI } from '@/api/api';
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+
+export default function useSearchToursAPI(url: string): UseQueryResult<SearchResponse, Error> {
+  return useQuery({
+    queryKey: ['tours-with-filter', url],
+    queryFn: () => searchToursAPI(url),
+    enabled: false,
+    refetchOnWindowFocus: false,
```

Comment:
I am not sure what this does but it looks like its a good thing to do so perhaps its alright to keep it this way. 
Just wanted to let you know ;)